File: projet1_android_java/app/src/main/java/com/example/tp_intro/MainActivity.java
package com.example.tp_intro;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    CheckBox chkAndro;
    CheckBox chkMicro;
    Button btnAffich;
    RadioGroup radSpecialityType;
    RadioButton radDSI;
    RadioButton radRSI;
    RadioButton radSEM;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        chkAndro = (CheckBox) findViewById(R.id.chkAndro);
        chkMicro = (CheckBox) findViewById(R.id.chkMicro);
        btnAffich = (Button) findViewById(R.id.btnAffich);
        radSpecialityType = (RadioGroup)
                findViewById(R.id.radioGroupSpecialiteType);
        radDSI = (RadioButton) findViewById(R.id.radDsi);
        radRSI = (RadioButton) findViewById(R.id.radRsi);
        radSEM =(RadioButton) findViewById(R.id.radSem);
        btnAffich.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("SuspiciousIndentation")
            @Override
            public void onClick(View v) {
                    String msg = "Spécialité ";
                    String option="club ";
                    if (chkAndro.isChecked())
                        option=option + chkAndro.getText() ;
                    if (chkMicro.isChecked())
                        option=option + chkMicro.getText();
                int radioId = radSpecialityType.getCheckedRadioButtonId();

                if (radSEM.getId() == radioId)
                    msg = msg+ radSEM.getText() ;
                else if (radRSI.isChecked())
                    msg = msg+ radRSI.getText() ;
                else if (radDSI.isChecked())
                    msg = msg+ radDSI.getText() ;

                Toast.makeText(getApplicationContext(), msg+" "+option, Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: app/src/main/java/com/example/recycleview/MainActivity.kt
package com.example.recycleview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity(), MyAdapter.OnItemClickListener {
    val values = arrayListOf<String>("item1", "item2" , "item3", "item4", "item5")
    private lateinit var recyclerView: RecyclerView
    private lateinit var manager: RecyclerView.LayoutManager
    private lateinit var myAdapter: RecyclerView.Adapter<*>
    private lateinit var bt :Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        manager = LinearLayoutManager(this)
        myAdapter = MyAdapter(values,this)
        recyclerView = findViewById<RecyclerView>(R.id.recyclerView).apply {
            layoutManager = manager
            adapter = myAdapter
        }
      bt=findViewById(R.id.bt)
        bt.setOnClickListener{
            var n:Int = values.size + 1
            values.add("item$n")
            myAdapter.notifyItemInserted(values.size)
        }
    }

    override fun onItemClick(position: Int) {
        Toast.makeText(this, "Item $position clicked",
            Toast.LENGTH_SHORT).show()
        values[position]="Clicked"
        myAdapter.notifyItemChanged(position)
    }

        }
